In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import scipy.stats as stats

In [2]:
#importing data into python environment
data=pd.read_csv('Sales_data_zone_wise.csv')

In [3]:
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [5]:
#checking for null values
data.isna().sum()

Month       0
Zone - A    0
Zone - B    0
Zone - C    0
Zone - D    0
dtype: int64

There are no null values in the data set

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Month     29 non-null     object
 1   Zone - A  29 non-null     int64 
 2   Zone - B  29 non-null     int64 
 3   Zone - C  29 non-null     int64 
 4   Zone - D  29 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.3+ KB


Data set have 29 rows and 5 columns

In [10]:
data.columns

Index(['Month', 'Zone - A', 'Zone - B', 'Zone - C', 'Zone - D'], dtype='object')

# 1. The mean sales generated by each zone.

In [17]:
data[['Zone - A', 'Zone - B', 'Zone - C', 'Zone - D']].mean()

Zone - A    1.540493e+06
Zone - B    1.755560e+06
Zone - C    1.772871e+06
Zone - D    1.842927e+06
dtype: float64

In [20]:
#another method
data.describe()

Zone - A      Zone - B      Zone - C      Zone - D
count  2.900000e+01  2.900000e+01  2.900000e+01  2.900000e+01
mean   1.540493e+06  1.755560e+06  1.772871e+06  1.842927e+06
std    2.619401e+05  1.683899e+05  3.331937e+05  3.750165e+05
min    1.128185e+06  1.527574e+06  1.237722e+06  1.234311e+06
25%    1.305972e+06  1.606010e+06  1.523308e+06  1.520406e+06
50%    1.534390e+06  1.740365e+06  1.767047e+06  1.854412e+06
75%    1.820196e+06  1.875658e+06  2.098463e+06  2.180416e+06
max    2.004480e+06  2.091194e+06  2.290580e+06  2.364132e+06

In [19]:
print('The mean sales of Zone-A = ',data['Zone - A'].mean())
print('The mean sales of Zone-B = ',data['Zone - B'].mean())
print('The mean sales of Zone-C = ',data['Zone - C'].mean())
print('The mean sales of Zone-D = ',data['Zone - D'].mean())

The mean sales of Zone-A =  1540493.1379310344
The mean sales of Zone-B =  1755559.5862068965
The mean sales of Zone-C =  1772871.0344827587
The mean sales of Zone-D =  1842926.7586206896


# 2.Total sales generated by all the zones for each month

In [21]:
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [80]:
data['Total Sales']=data.sum(axis=1)
data

Month  Zone - A  Zone - B  Zone - C  Zone - D  Total Sales
0    Month - 1   1483525   1748451   1523308   2267260      7022544
1    Month - 2   1238428   1707421   2212113   1994341      7152303
2    Month - 3   1860771   2091194   1282374   1241600      6475939
3    Month - 4   1871571   1759617   2290580   2252681      8174449
4    Month - 5   1244922   1606010   1818334   1326062      5995328
5    Month - 6   1534390   1573128   1751825   2292044      7151387
6    Month - 7   1820196   1992031   1786826   1688055      7287108
7    Month - 8   1625696   1665534   2161754   2363315      7816299
8    Month - 9   1652644   1873402   1755290   1422059      6703395
9   Month - 10   1852450   1913059   1754314   1608387      7128210
10  Month - 11   2004480   1740365   1317869   1970069      7032783
11  Month - 12   1537539   1875658   1237722   1460165      6111084
12  Month - 13   1254939   1588473   1348629   1733383      5925424
13  Month - 14   1821799   1844081   1617376   1872259      7155515
14  Month - 15   1128185   1804613   1767047   1234311      5934156
15  Month - 16   1491534   1542053   1882159   1590913      6506659
16  Month - 17   1182110   1552388   2234469   2180416      7149383
17  Month - 18   1770324   1699213   1249821   2364132      7083490
18  Month - 19   1181635   1959268   2009168   1821882      6971953
19  Month - 20   1305972   1559670   2107949   2151008      7124599
20  Month - 21   1570152   2013615   1540016   2265814      7389597
21  Month - 22   1481619   1527574   2255729   2295079      7560001
22  Month - 23   1887694   1992155   1348387   1459683      6687919
23  Month - 24   1880820   1752873   2098463   2052591      7784747
24  Month - 25   1256333   1622671   1521792   1695122      6095918
25  Month - 26   1422853   1715465   1853636   1520406      6512360
26  Month - 27   1384426   1983163   1611169   1289160      6267918
27  Month - 28   1616640   1547991   2128022   2178267      7470920
28  Month - 29   1310654   1660092   1947119   1854412      6772277

In [81]:
data[['Month','Total Sales']]

Month  Total Sales
0    Month - 1      7022544
1    Month - 2      7152303
2    Month - 3      6475939
3    Month - 4      8174449
4    Month - 5      5995328
5    Month - 6      7151387
6    Month - 7      7287108
7    Month - 8      7816299
8    Month - 9      6703395
9   Month - 10      7128210
10  Month - 11      7032783
11  Month - 12      6111084
12  Month - 13      5925424
13  Month - 14      7155515
14  Month - 15      5934156
15  Month - 16      6506659
16  Month - 17      7149383
17  Month - 18      7083490
18  Month - 19      6971953
19  Month - 20      7124599
20  Month - 21      7389597
21  Month - 22      7560001
22  Month - 23      6687919
23  Month - 24      7784747
24  Month - 25      6095918
25  Month - 26      6512360
26  Month - 27      6267918
27  Month - 28      7470920
28  Month - 29      6772277

# 3.Checking whether all the zones generate the same amount of sales.

# We will use one-way ANOVA 

Null Hypothesis,H0 : All zones generate same amount of sales
    
Alternate Hypothesis,H1 : Sales of atleast one zone is different

In [6]:
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [7]:
data.shape

(29, 5)

In [25]:
alpha=0.05

In [8]:
#numerator degrees of freedom (k-1) = (4-1) = 3
dfn=3
#denominator degrees of freedom (N-k)= (29*4-4) = 112

dfd=112

In [14]:
#critical f- value
f_critical=stats.f.ppf(q=.95,dfn=3,dfd=112)
print('Critical F - value = ',f_critical)

Critical F - value =  2.6856434512366034


In [11]:
f_value,p_value=stats.f_oneway(data['Zone - A'],data['Zone - B'],data['Zone - C'],data['Zone - D'])

In [13]:
print('Calculated F - value = ',f_value)

print('P - value = ',p_value)

F - value =  5.672056106843581
P - value =  0.0011827601694503335


# Another method using  STATSMODELS

In [18]:

import statsmodels.api as sm
from statsmodels.formula.api import ols

In [16]:
df_melt = pd.melt(data.reset_index(), id_vars=['index'], value_vars=['Zone - A', 'Zone - B', 'Zone - C', 'Zone - D'])
df_melt

index  variable    value
0        0  Zone - A  1483525
1        1  Zone - A  1238428
2        2  Zone - A  1860771
3        3  Zone - A  1871571
4        4  Zone - A  1244922
..     ...       ...      ...
111     24  Zone - D  1695122
112     25  Zone - D  1520406
113     26  Zone - D  1289160
114     27  Zone - D  2178267
115     28  Zone - D  1854412

[116 rows x 3 columns]

In [26]:
model = ols('value ~ C(variable)', data=df_melt).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
anova_table

sum_sq     df         F    PR(>F)
C(variable)  1.483058e+12    3.0  5.672056  0.001183
Residual     9.761449e+12  112.0       NaN       NaN

In [27]:
f_critical>f_value

False

# Interpretation:

The F test statistic is 5.6720 and the corresponding p-value is 0.001183. Since the p-value is  less than .05 and Caculated F-value is greater than critical F-value we are forced to reject the null hypothesis (H0).

We can infer that mean of atleast one column is different ,ie, the sales varies with respect to the zones